packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/java/siw.md

```diff
@@ -0,0 +1,124 @@
+
+```java
+  String issuer = oktaOAuth2Properties.getIssuer();
+  // the widget needs the base url, just grab the root of the issuer
+  String orgUrl = new URL(new URL(issuer), "/").toString();
+
+  ModelAndView mav = new ModelAndView("login");
+  mav.addObject(STATE, state);
+  mav.addObject(OTP, otp);
+  mav.addObject(SCOPES, oktaOAuth2Properties.getScopes());
+  mav.addObject(OKTA_BASE_URL, orgUrl);
+  mav.addObject(OKTA_CLIENT_ID, oktaOAuth2Properties.getClientId());
+  mav.addObject(INTERACTION_HANDLE, idxClientContext.getInteractionHandle());
+  mav.addObject(CODE_VERIFIER, idxClientContext.getCodeVerifier());
+  mav.addObject(CODE_CHALLENGE, idxClientContext.getCodeChallenge());
+  mav.addObject(CODE_CHALLENGE_METHOD, CODE_CHALLENGE_METHOD_VALUE);
+  mav.addObject(REDIRECT_URI,
+    request.getScheme() + "://" + request.getServerName() + ":" +
+    request.getServerPort() + request.getContextPath() +
+    "/authorization-code/callback");
+  mav.addObject(ISSUER_URI, issuer);
+  session.setAttribute(CODE_VERIFIER, idxClientContext.getCodeVerifier());
+  return mav;
+}
+```
+
+You can pass the values to the widget using the widget's `config` object.
+
+```html
+  <div id="sign-in-widget"></div>
+
+  <script th:inline="javascript">
+      /*<![CDATA[*/
+
+      var config = {};
+      config.baseUrl = /*[[${oktaBaseUrl}]]*/ 'https://{yourOktaDomain}';
+      config.clientId = /*[[${oktaClientId}]]*/ '{clientId}';
+      config.otp = /*[[${otp}]]*/ '{otp}';
+      config.redirectUri = /*[[${redirectUri}]]*/ '{redirectUri}';
+      config.useInteractionCodeFlow = true;
+      config.interactionHandle = /*[[${interactionHandle}]]*/ '{interactionHandle}';
+      config.codeChallenge = /*[[${codeChallenge}]]*/ '{codeChallenge}';
+      config.codeChallengeMethod = /*[[${codeChallengeMethod}]]*/ '{codeChallengeMethod}';
+      config.redirect = 'always';
+      config.authParams = {
+          issuer: /*[[${issuerUri}]]*/ '{issuerUri}',
+          pkce: true,
+          state: /*[[${state}]]*/ '{state}' || false,
+          nonce: /*[[${nonce}]]*/ '{nonce}',
+          scopes: /*[[${scopes}]]*/ '[scopes]',
+      };
+
+      new OktaSignIn(config).showSignInAndRedirect(
+          { el: '#sign-in-widget' },
+          function (res) {}
+      );
+      /*]]>*/
+  </script>
+```
+
+### 3: Displays password reset prompt and complete password recovery flow
+
+After the widget is loaded, it checks whether `state` and `otp` are valid with the Okta server. Assuming they are, either the following reset page or a prompt appears for the user to enter the OTP. After the user enters the OTP, the reset page appears. The user continues the password recovery flow described in the [User password recovery guide](/docs/guides/oie-embedded-sdk-use-case-pwd-recovery-mfa/java/main/).
+
+<div class="common-image-format">
+
+![Screenshot of password reset page](/img/advanced-use-cases/java-custom-pwd-recovery-custom-siw-reset-pwd-page.png "Password Reset Page")
```

Comment:
This image is quite big compared to the sequence diagrams. Is there any way to reduce the size? Or maybe not display this image if it's not required?